File: Solutions/11.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int max_area = 0;
        int left = 0;
        int right = height.size() - 1;

        while (left < right)
        {
            if (height[left] < height[right])
            {
                max_area = max(max_area, (right - left) * (height[left]));
                left++;
            }else
            {
                max_area = max(max_area, (right - left) * (height[right]));
                right--;
            }
            
        }
        return max_area;
    }
};

int main()
{
    return 0;
}

File: Solutions for Python/11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1

        ans = -1

        while l < r:
            ans = max(ans, (r - l)*min(height[l],height[r]))
            if height[l] <= height[r]:
                l += 1
            else:
                r -= 1
        return ans


if __name__ == "__main__":
    pass
